scripts/release_support/changelog.sh

```diff
@@ -88,47 +89,52 @@ for component in "${!COMPONENT_LIST[@]}"
+                start_hash=$(grep "$component-" start_build_manifest.txt | head -1 | awk -F['_'] '{print $2}' | cut -d. -f1 |  sed 's/git//g'); echo "$start_hash"
+                target_hash=$(grep "$component-" target_build_manifest.txt | head -1 | awk -F['_'] '{print $2}' | cut -d. -f1 |  sed 's/git//g'); echo "$target_hash"
+        elif [ "$component" == "cortx-csm_agent" ] || [ "$component" == "cortx-csm_web" ]; then
+                start_hash=$(grep "$component-" start_build_manifest.txt | head -1 | awk -F['_'] '{print $3}' |  cut -d. -f1); echo "$start_hash"
+                target_hash=$(grep "$component-" target_build_manifest.txt | head -1 | awk -F['_'] '{print $3}' |  cut -d. -f1); echo "$target_hash"
+        elif [ "$component" == "cortx-provisioner" ] || [ "$component" == "cortx-rgw-integration" ] ; then
+                start_hash=$(grep "$component-" start_build_manifest.txt | tail -1 | awk -F['_'] '{print $2}' | sed 's/git//g' | cut -d. -f1); echo "$start_hash"
+                target_hash=$(grep "$component-" target_build_manifest.txt | tail -1 | awk -F['_'] '{print $2}' | sed 's/git//g' | cut -d. -f1); echo "$target_hash"
+        elif [ "$component" == "ceph-base" ]; then
+                start_hash=$(grep "$component-" start_build_manifest.txt | awk -F['-'] '{print $5}'  | cut -d. -f2 | sed s/g//g); echo "$start_hash"
+                target_hash=$(grep "$component-" target_build_manifest.txt | awk -F['-'] '{print $5}'  | cut -d. -f2 | sed s/g//g); echo "$target_hash"
+        else
+                start_hash=$(grep "$component-" start_build_manifest.txt | head -1 | awk -F['_'] '{print $2}' | sed 's/git//g'|cut -d. -f1); echo "$start_hash"
+                target_hash=$(grep "$component-" target_build_manifest.txt | head -1 | awk -F['_'] '{print $2}' | sed 's/git//g'|cut -d. -f1); echo "$target_hash"
+        fi
+        
+        pushd "$dir" || exit
+                commit_sha="$(git log "$start_hash..$target_hash" --oneline --pretty=format:"%h")";
+                if [ "$commit_sha" ]; then
+                        for commit in $commit_sha; do
+                                original_commit_message=$(git log --oneline -n 1 "$commit" --pretty=format:"%s")
+                                filtered_commit_message=$(sed -e 's/([^()]*)//g' <<< $original_commit_message)
+                                repo_name=$(awk -F"[/.]" '{print $6}' <<< ${COMPONENT_LIST[$component]})
+                                pr_number=$(curl -s -H "Accept: application/json" -H "Authorization: token $ACCESS_TOKEN" https://api.github.com/repos/seagate/"$repo_name"/commits/"$commit"/pulls | jq '.[].number')
+                                pr_url=$(curl -s -H "Accept: application/json" -H "Authorization: token $ACCESS_TOKEN" https://api.github.com/repos/seagate/"$repo_name"/commits/"$commit"/pulls | jq '.[].html_url' | sed "s/\"//g")
+                                if [ "$pr_number" ] && [ "$pr_url" ]; then
```

Comment:
We can use `awk` to fetch PR number from PR URL instead of again calling GitHub API
```
[root@ssc-vm-rhev4-0707 ~]# pr_url=$(curl -s -H 'Accept: application/json' https://api.github.com/repos/seagate/cortx-manager/commits/b664d7e/pulls | jq '.[].html_url' | sed "s/\"//g")
[root@ssc-vm-rhev4-0707 ~]# pr_number=$(echo $pr_url | awk -F[/] '{print $NF}')
[root@ssc-vm-rhev4-0707 ~]# echo $pr_url
https://github.com/Seagate/cortx-manager/pull/866
[root@ssc-vm-rhev4-0707 ~]# echo $pr_number
866
[root@ssc-vm-rhev4-0707 ~]#
```

Reply:
Updated and tested - https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/gc-space/job/changelog-test/21/